overlord/hookstate/ctlcmd/set.go

```diff
@@ -99,6 +102,27 @@ func (s *setCommand) Execute(args []string) error {
 	if snap != "" {
 		return fmt.Errorf(`"snapctl set %s" not supported, use "snapctl set :%s" instead`, s.Positional.PlugOrSlotSpec, parts[1])
 	}
+
+	if s.View {
+		requests := make(map[string]interface{}, len(s.Positional.ConfValues))
+		for _, conf := range s.Positional.ConfValues {
+			splitConf := strings.SplitN(conf, "=", 2)
+			if len(splitConf) == 1 && strings.HasSuffix(conf, "!") {
+				key := strings.TrimSuffix(conf, "!")
+				requests[key] = nil
+				continue
+			}
+
+			if len(splitConf) != 2 {
+				return fmt.Errorf("cannot set %s plug: must set field with \"=\" or unset with \"!\"", s.Positional.PlugOrSlotSpec)
+			}
```

Comment:
couldn't a helper be shared with the config code for some of this? otoh even if not shared maybe the loop/part the loop should be in a helper?